### ISSUE TITLE
Feature Request: register the number of particles of each kind

### ISSUE THREAD

**Issue by MatthieuSchaller** [SWIFTSIM/velociraptor-python#66]
Is that possible?  It would let us fix the "zoom" problem bu ignoring structures dominated by background particles.

Other option, store the mass by type but also for the background particles.

**Comment by EvgeniiChaikin** [comment]
Yes, and it is actually rather easy to do - I've just checked.

If  VR is compiled with `ZOOM_SIM` - which is of course true for zooms - then in the output catalog, there is a field `Mass_interloper` telling us the total mass of the background particles.

We need to decide though how exactly we want to get rid of the problematic haloes. 

E.g.,

- It could be done as part of another post-processing script (which would generate a new halo catalogue devoid of the problematic haloes)
- Or, **which I think is the best way**, we could just apply a mask to the SMHM plots. I really like this idea because in the auto plotter all the code for applying masks has already been written by Josh a long time ago, so adding a new mask is very straightforward.

**Comment by MatthieuSchaller** [comment]
Yes, a mask removing haloes that are heavily contaminated would be best. If the corresponding contamination count is already picked up by the VR python module then it's all easy to do indeed in the pipeline code.

**Comment by james-trayford** [comment]
see PRs https://github.com/SWIFTSIM/pipeline-configs/pull/130, https://github.com/SWIFTSIM/pipeline/pull/25, https://github.com/SWIFTSIM/velociraptor-python/pull/68
